services/auth/interface.go

```diff
@@ -28,10 +30,10 @@ type SingleSignOn interface {
 	// give chance to the plugin to free any allocated resources
 	Free() error
 
-	// IsEnabled checks if the current SSO method has been enabled in settings.
+	// IsEnabled checks if the current auth method has been enabled in settings.
```

Comment:
We should think about whether we need this.

 It could be rolled into the verifyauthdata function quite easily thus reducing complexity in the functions that call it, or (likely better) the methods groups could be constructed with only enabled methods enabled 

Once we have the groups peppered within the routes we should then simplify the rest of these methods to remove the URL testing within methods.